internal/provider/provider.go

```diff
@@ -209,6 +209,7 @@ func Provider(config *Config) plugin.ProviderFunc {
 				"scaleway_secret":                              secret.ResourceSecret(),
 				"scaleway_secret_version":                      secret.ResourceVersion(),
 				"scaleway_tem_domain":                          tem.ResourceDomain(),
```

Comment:
No, because to validate the domain, we have to make records that depend on the scaleway_tem_domain resource. See => https://registry.terraform.io/providers/scaleway/scaleway/latest/docs/resources/tem_domain#add-the-required-records-to-your-dns-zone. We must wait for the records on the domain before the domain_tem verification.